types/lib/legacy/legacy-element-mixin.d.ts

```diff
@@ -0,0 +1,620 @@
+     * @returns Returns the given `object` with its prototype set
+     * to the given `prototype` object.
+     */
+    chainObject(object: Object|null, prototype: Object|null): Object|null;
+
+    /**
+     * Calls `importNode` on the `content` of the `template` specified and
+     * returns a document fragment containing the imported content.
+     *
+     * @param template HTML template element to instance.
+     * @returns Document fragment containing the imported
+     *   template content.
+     */
+    instanceTemplate(template: HTMLTemplateElement|null): DocumentFragment|null;
+
+    /**
+     * Dispatches a custom event with an optional detail value.
+     *
+     * @param type Name of event type.
+     * @param detail Detail value containing event-specific
+     *   payload.
+     * @param options Object specifying options.  These may include:
+     *  `bubbles` (boolean, defaults to `true`),
+     *  `cancelable` (boolean, defaults to false), and
+     *  `node` on which to fire the event (HTMLElement, defaults to `this`).
+     * @returns The new event that was fired.
+     */
+    fire(type: string, detail?: any, options?: any): Event|null;
+
+    /**
+     * Convenience method to add an event listener on a given element,
+     * late bound to a named method on this element.
+     *
+     * @param node Element to add event listener to.
+     * @param eventName Name of event to listen for.
+     * @param methodName Name of handler method on `this` to call.
+     */
+    listen(node: Element|null, eventName: string, methodName: string): void|null;
+
+    /**
+     * Convenience method to remove an event listener from a given element,
+     * late bound to a named method on this element.
+     *
+     * @param node Element to remove event listener from.
+     * @param eventName Name of event to stop listening to.
+     * @param methodName Name of handler method on `this` to not call
+     *        anymore.
+     */
+    unlisten(node: Element|null, eventName: string, methodName: string): void|null;
+
+    /**
+     * Override scrolling behavior to all direction, one direction, or none.
+     *
+     * Valid scroll directions:
+     *   - 'all': scroll in any direction
+     *   - 'x': scroll only in the 'x' direction
+     *   - 'y': scroll only in the 'y' direction
+     *   - 'none': disable scrolling for this node
+     *
+     * @param direction Direction to allow scrolling
+     * Defaults to `all`.
+     * @param node Element to apply scroll direction setting.
+     * Defaults to `this`.
+     */
+    setScrollDirection(direction?: string, node?: Element|null): void|null;
+
+    /**
+     * Convenience method to run `querySelector` on this local DOM scope.
+     *
+     * This function calls `Polymer.dom(this.root).querySelector(slctr)`.
+     *
+     * @param slctr Selector to run on this local DOM scope
+     * @returns Element found by the selector, or null if not found.
+     */
+    $$(slctr: string): Element|null;
```

Comment:
That would be cool. I'd need a way to directly override the TS definition though, because there'd be no way to define that with a JSDoc annotation. Will probably punt for now. Might end up writing an override mechanism later.